launcher/launcher/main.go

```diff
@@ -42,76 +44,81 @@ var rcMessage = map[int]string{
-	log.SetOutput(os.Stdout)
 	defer func() {
 		os.Exit(exitCode)
 	}()
 
-	serialConsole, err := os.OpenFile("/dev/console", os.O_WRONLY, 0)
+	logger, err = logging.NewLogger(ctx)
 	if err != nil {
-		logger.Printf("failed to open serial console for writing: %v\n", err)
+		log.Default().Printf("failed to initialize logging")
 		exitCode = failRC
-		logger.Printf("%s, exit code: %d (%s)\n", exitMessage, exitCode, rcMessage[exitCode])
+		log.Default().Printf("%s, exit code: %d (%s)\n", exitMessage, exitCode, rcMessage[exitCode])
 		return
 	}
-	defer serialConsole.Close()
-	logger.SetOutput(io.MultiWriter(os.Stdout, serialConsole))
+	defer logger.Close()
 
-	logger.Println(welcomeMessage)
-	logger.Printf("Build commit: %s\n", BuildCommit)
+	logger.Info(welcomeMessage, "build_commit", BuildCommit)
 
 	if err := verifyFsAndMount(); err != nil {
-		logger.Printf("failed to verify filesystem and mounts: %v\n", err)
+		logger.Error(fmt.Sprintf("failed to verify filesystem and mounts: %v\n", err))
 		exitCode = rebootRC
-		logger.Printf("%s, exit code: %d (%s)\n", exitMessage, exitCode, rcMessage[exitCode])
+		logger.Error(exitMessage, "exit_code", exitCode, "exit_msg", rcMessage[exitCode])
 		return
 	}
 
 	if err := os.MkdirAll(launcherfile.HostTmpPath, 0744); err != nil {
-		logger.Printf("failed to create %s: %v", launcherfile.HostTmpPath, err)
+		logger.Error(fmt.Sprintf("failed to create %s: %v", launcherfile.HostTmpPath, err))
 	}
 
 	// Get RestartPolicy and IsHardened from spec
 	mdsClient = metadata.NewClient(nil)
 	launchSpec, err := spec.GetLaunchSpec(ctx, logger, mdsClient)
 	if err != nil {
-		logger.Printf("failed to get launchspec, make sure you're running inside a GCE VM: %v\n", err)
+		logger.Error(fmt.Sprintf("failed to get launchspec, make sure you're running inside a GCE VM: %v", err))
 		// if cannot get launchSpec, exit directly
 		exitCode = failRC
-		logger.Printf("%s, exit code: %d (%s)\n", exitMessage, exitCode, rcMessage[exitCode])
+		logger.Error(exitMessage, "exit_code", exitCode, "exit_msg", rcMessage[exitCode])
 		return
 	}
 
 	defer func() {
 		// Catch panic to attempt to output to Cloud Logging.
 		if r := recover(); r != nil {
-			logger.Println("Panic:", r)
+			logger.Error(fmt.Sprintf("Panic: %v", r))
 			exitCode = 2
 		}
 		msg, ok := rcMessage[exitCode]
 		if ok {
-			logger.Printf("%s, exit code: %d (%s)\n", exitMessage, exitCode, msg)
+			logger.Info(exitMessage, "exit_code", exitCode, "exit_msg", msg)
 		} else {
-			logger.Printf("%s, exit code: %d\n", exitMessage, exitCode)
+			logger.Info(exitMessage, "exit_code", exitCode)
 		}
 	}()
-	if err = startLauncher(ctx, launchSpec, serialConsole); err != nil {
-		logger.Println(err)
+	if err = startLauncher(launchSpec, logger.SerialConsoleFile()); err != nil {
+		logger.Error(err.Error())
 	}
 
+	workloadDuration := time.Since(start)
+	logger.Info("Workload completed",
+		"workload", launchSpec.ImageRef,
+		"latency_sec", workloadDuration.Seconds(),
```

Comment:
not sure if "latency_sec" is set somewhere else, but maybe "elapsed_sec" is more suitable in the context?